src/logger/loggerApi.c

```diff
@@ -1712,3 +1724,59 @@ int api_get_wifi_client_cfg(struct Serial *serial, const jsmntok_t *json)
 
         return API_SUCCESS_NO_RETURN;
 }
+
+int api_set_wifi_ap_cfg(struct Serial *s, const jsmntok_t *json)
+{
+        LoggerConfig *lc = getWorkingLoggerConfig();
+        struct wifi_ap_cfg *cfg = &lc->ConnectivityConfigs.wifi.ap;
+
+        struct wifi_ap_cfg tmp_cfg;
+        memcpy(&tmp_cfg, cfg, sizeof(struct wifi_ap_cfg));
+
+        setBoolValueIfExists(json, "active", &tmp_cfg.active);
+        setStringValueIfExists(json, "ssid", tmp_cfg.ssid,
+                               ARRAY_LEN(tmp_cfg.ssid));
+        setStringValueIfExists(json, "passwd", tmp_cfg.password,
+                               ARRAY_LEN(tmp_cfg.password));
+        setIntValueIfExists(json, "channel", (int*) &tmp_cfg.channel);
+
+
+        char enc_str[12];
+        setStringValueIfExists(json, "encryption", enc_str,
+                               ARRAY_LEN(enc_str));
+        tmp_cfg.encryption = wifi_api_get_encryption_enum_val(enc_str);
+
+        if (!wifi_validate_ap_config(&tmp_cfg)) {
+                pr_info("Invalid Wifi AP config given\r\n");
+                return API_ERROR_PARAMETER;
+        }
+
+        /* Copy the validated config to our real config */
+        memcpy(cfg, &tmp_cfg, sizeof(struct wifi_ap_cfg));
+
+        /* Inform the Wifi device that settings may have changed */
+        wifi_update_ap_config(cfg);
+
+        return API_SUCCESS;
+}
+
+int api_get_wifi_ap_cfg(struct Serial *serial, const jsmntok_t *json)
+{
+        LoggerConfig *lc = getWorkingLoggerConfig();
+        const struct wifi_ap_cfg *cfg = &lc->ConnectivityConfigs.wifi.ap;
+
+        json_objStart(serial);
+        json_objStartString(serial, "wifi_ap_cfg");
+
+        json_bool(serial, "active", cfg->active, 1);
+        json_string(serial, "ssid", cfg->ssid,1);
+        json_string(serial, "passwd", cfg->password,1);
+        json_int(serial, "channel", cfg->channel,1);
+        json_string(serial, "encryption",
+                    wifi_api_get_encryption_str_val(cfg->encryption), 0);
+
+        json_objEnd(serial, 0);
+        json_objEnd(serial, 0);
```

Comment:
The double is likely to close the json object so it is balanced. I would
double check .

Brent Picasso
Autosport Labs
Technology for Race and Street
On May 12, 2016 7:37 AM, "Andrew Stiegmann" <notifications@github.com>
wrote:

> In src/logger/loggerApi.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/603#discussion_r63031507>
> :
>
> > +{
> > +        LoggerConfig *lc = getWorkingLoggerConfig();
> > +        const struct wifi_ap_cfg *cfg = &lc->ConnectivityConfigs.wifi.ap;
> > +
> > +        json_objStart(serial);
> > +        json_objStartString(serial, "wifi_ap_cfg");
> > +
> > +        json_bool(serial, "active", cfg->active, 1);
> > +        json_string(serial, "ssid", cfg->ssid,1);
> > +        json_string(serial, "passwd", cfg->password,1);
> > +        json_int(serial, "channel", cfg->channel,1);
> > +        json_string(serial, "encryption",
> > +                    wifi_api_get_encryption_str_val(cfg->encryption), 0);
> > +
> > +        json_objEnd(serial, 0);
> > +        json_objEnd(serial, 0);
>
> I think I may have a copy/pasta error on line 1768 and I added that to fix
> that issue. That was bugging me late at night. Will fix.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/603/files/9a96dfe939dfb3ce683be929f40c698e9c54a437#r63031507>
>